### ISSUE TITLE
Failed login using docker_registry does not trigger converge failure

### ISSUE THREAD

**Issue by mrdziuban** [bflad/chef-docker#398]
When using an invalid password with `docker_registry`, the login fails, but the converge finishes without displaying any errors. Here's what my default recipe looks like:

```
docker_service 'default' do
  version '1.8.1'
  action [:create, :start]
end

docker_registry 'https://index.docker.io/v1/' do
  username 'mrdziuban'
  password 'INVALID_PASSWORD'
  email 'email@email.com'
end
```

I can confirm that the login failed by looking in `/var/log/upstart/docker.log`, which shows:

```
time="2015-08-26T14:53:35.606393896Z" level=info msg="POST /v1.16/auth" 
time="2015-08-26T14:53:36.314017119Z" level=error msg="Handler for POST /auth returned error: Wrong login/password, please try again" 
time="2015-08-26T14:53:36.314162672Z" level=error msg="HTTP Error" err="Wrong login/password, please try again" statusCode=401 
```

So when I try to pull a private image from the registry, I get an error because it can't find the repo:

```
Error: image org/image-name:latest not found
```

**Comment by skmichaelson** [comment]
It looks like the exception is rescued and never re-raised if the retry limit was reached.

**Comment by someara** [comment]
FIxed in 1.0.19
